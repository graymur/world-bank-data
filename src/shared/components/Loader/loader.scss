@import "~style/lib";

.loader {
	font-size: 3rem;
	text-align: center;
}
