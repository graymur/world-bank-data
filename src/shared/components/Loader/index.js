import React from 'react';
import PropTypes from 'prop-types';

import './loader.scss';

export default class Loader extends React.PureComponent {
	static propTypes = {
		size: PropTypes.number,
		className: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.arrayOf(PropTypes.string)
		])
	};

	defaultProps = {
		className: []
	};

	constructor(props) {
		super(props);

		let {className} = this.props;

		if (!Array.isArray(className)) {
			className = [className];
		}

		className.push('loader');

		this.className = className;
	}

	render() {
		let size = this.props.size || 50;
		let borderWidth = Math.max(2, Math.ceil(size * 0.2));

		const style = {
			width: `${size}px`,
			height: `${size}px`,
			border: `${borderWidth}px solid #f3f3f3`,
			borderTop: `${borderWidth}px solid #3498db`
		};

		return (
			<span className={this.className.join(' ')} style={style}/>
		);
	}
}
