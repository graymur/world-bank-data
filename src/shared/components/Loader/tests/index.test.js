/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../index';

test('should render an <span> tag', () => {
	const component = renderer.create(<Loader />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('should have correct width, height and border', () => {
	const component = renderer.create(<Loader size={100}/>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('should accept additional class name', () => {
	const component = renderer.create(<Loader className={'test-class-name'}/>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
