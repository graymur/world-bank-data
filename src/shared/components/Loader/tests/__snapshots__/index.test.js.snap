// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept additional class name 1`] = `
<span
  className="test-class-name loader"
  style={
    Object {
      "border": "10px solid #f3f3f3",
      "borderTop": "10px solid #3498db",
      "height": "50px",
      "width": "50px",
    }
  }
/>
`;

exports[`should have correct width, height and border 1`] = `
<span
  className="loader loader"
  style={
    Object {
      "border": "20px solid #f3f3f3",
      "borderTop": "20px solid #3498db",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;

exports[`should render an <span> tag 1`] = `
<span
  className="loader"
  style={
    Object {
      "border": "10px solid #f3f3f3",
      "borderTop": "10px solid #3498db",
      "height": "50px",
      "width": "50px",
    }
  }
/>
`;
